app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileEditFragmentTest.kt

```diff
@@ -158,6 +166,36 @@ class ProfileEditFragmentTest {
     }
   }
 
+  @Test
+  fun testProfileEdit_startWithUserProfile_clickProfileDeletionButton_deleteSnackbarIsVisible() {
+    launchFragmentTestActivity(internalProfileId = 1).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.profile_edit_delete_dialog_positive))
+        .inRoot(isDialog())
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withText(R.string.profile_edit_delete_success)))
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testProfileEdit_startWithUserProfile_AfterDeleteSnackbar_profileActivityIsShownAgain() {
+    launchFragmentTestActivity(internalProfileId = 1).use {
+      onView(withId(R.id.profile_delete_button)).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.profile_edit_delete_dialog_positive))
+        .inRoot(isDialog())
+        .perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(allOf(withText(R.string.profile_edit_delete_success)))
+        .check(matches(isDisplayed()))
+      testCoroutineDispatchers.runCurrent()
+      intended(hasComponent(ProfileListActivity::class.java.name))
+    }
+  }
+
```

Comment:
These tests are failing on CI. Are they actually passing on your local?

Reply:
Yes @adhiamboperes it is passing on my local

Reply:
And for this https://github.com/oppia/oppia-android/pull/4887#discussion_r1269465287, I have removed it from my PR.

Reply:
Okay

Reply:
@adhiamboperes I think the tests are failing because snackbar is hiding before this is asserted, wdyt?

Reply:
I am not so sure. Could you please try out my suggestions above and see if it works?

Reply:
Okay